### ISSUE TITLE
Scripts hitting max quota very quickly.

### ISSUE THREAD

**Issue by McDunkinY** [INPStarfall/Starfall#405]
Before a ( recent? ) update, all of my client-sided scripts worked correctly, but now they crash as soon as I spawn them. For example, I have a fully client-sided 3D library which I load onto another script and it crashes with something as minimal as drawing like ten rectangles. The time buffer values are still the default values so none of that has changed. I've plenty of years of experience with Lua so it's not a logic issue / bad programming. You can't draw polygons without the processor crashing a few seconds after. I've even ran an empty processor with no operations other than the usage of the hook for mouse coords and it crashes a while after. I understand this may have something to do with garbage collection causing a sudden spike, but it's really gotten to the point where I can't use it. I can't speak for performance server wise since I instantly return if it's not being ran on the client realm.

Please let me know if something you guys changed might have caused this.

Thank you for your time.

**Comment by Xandaros** [comment]
Likely caused by #374. Nothing we can do about it, though. Before that, the cpuTime counting got accidentally reset every now and then, allowing you to use a lot more time than you actually had. We recently changed the default value to 1.5x what it was before in #403 to counteract this a little.
Make sure you are actually using the new default. If that is not enough, all you can do is increase the limit on your end.

Should you find a value that works better and still works with lots of players, let us know and we might think about raising the default further. Keep in mind, that this value doesn't work for every server and especially in single player can be raised by quite a bit without issue.

**Comment by awilliamson** [comment]
I don't believe we changed the client convar in that update, only the server. It wasn't reported to us at the time that screens were hitting quota. Please do feel free to suggest a new value as @Xandaros mentioned.

It's hard to get these values correct as it all depends on the 'strength' of the server, how many people are currently playing, how much Lua is being ran usually, how many chips each player has. There's a risk that if the value is set too high that it will negatively impact players, and as you've seen, too low and things don't work.

**Comment by Jazzelhawk** [comment]
I should mention that when I was making the new editor I found that convars only change when you set them from Lua or through the console and not when you create them as in #403. This means that although the new default is higher, all servers which had the previous default will have to manually change their convars.
